eth_tester/main.py

```diff
@@ -64,6 +68,29 @@ def get_default_fork_blocks():
     }
 
 
+def handle_auto_mining(func):
+    @functools.wraps(func)
+    def func_wrapper(self, *args, **kwargs):
+        if not self.auto_mine_transactions:
+            snapshot = self.take_snapshot()
+
+        transaction_hash = func(self, *args, **kwargs)
+
+        if self.auto_mine_transactions:
+            self.mine_block()
+        else:
+            pending_transaction = self.get_transaction_by_hash(transaction_hash)
+            # Remove any pending transactions with the same nonce
+            self._pending_transactions = [tx for tx in self._pending_transactions
```

Comment:
Can we unroll this comprehension into a small stand alone utility function for better readability as well as easier stand-alone testing.

```python
@to_tuple
def (tx, pending_transactions):
    for transaction in pending_transactions:
        ...
        yield transaction
```

Something that's got the same logic, but unrolled into an easier to read code body.